sdk/storage/azure-storage-blob/azure/storage/blob/_shared/base_client_async.py

```diff
@@ -121,6 +123,11 @@ def _create_pipeline(self, credential, **kwargs):
         try:
             if response.status_code not in [202]:
                 raise HttpResponseError(response=response)
-            return response.parts()  # Return an AsyncIterator
+            parts = response.parts() # Return an AsyncIterator
+            if raise_on_any_failure:
+                failures = [p for p in parts if p.status_code not in [202]]
```

Comment:
The comment above says that `parts` is an async iterator?
In that case this list comprehension wont work - it would need to be:
```
failures = []
async for part in parts:
    if part.status_code not in [202]:
        failures.append(part)
```
Kinda painful I know... but async list comprehensions weren't supported till 3.6....